### PR TITLE
Implement fn:default-language#0

### PR DESCRIPTION
**Pull request by alanpaxton**
### Description:

This is an implementation of the XQuery 3 default-language function missing from eXist-db.

### Reference:

This has been reported as part of https://github.com/eXist-db/exist/issues/3635
The function is defined in https://www.w3.org/TR/xpath-functions-31/#func-default-language

### Type of tests:

A fnLanguage.xqm test is provided which confirms that a default language is returned and that it is one of the known languages. The language is fetched in Java from the underlying default Locale, and will naturally vary depending where the tests are run.

A defaultLanguage() test is added to XQueryFunctionsTest.java which confirms that running the method in the embedded server returns the same language as directly accessing the default locale through Java.

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FnDefaultLanguage.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    "during the evaluation of a query or transformation in which " +
+                    "fn:default-language() is executed.",
+            FunctionDSL.returns(Type.LANGUAGE, Cardinality.EXACTLY_ONE, "the default language within query execution time span"));
+
+    public FnDefaultLanguage(final XQueryContext context) {
+        super(context, FS_DEFAULT_LANGUAGE);
+    }
+
+    public Sequence eval(final Sequence[] args, final Sequence contextSequence) throws XPathException {
+        if (context.getProfiler().isEnabled()) {
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/functions/fn/FnDefaultLanguage.java:49]: basic function already implements this profiler

**Review by dizzzz — APPROVED** [review]

**Review by dizzzz — CHANGES_REQUESTED** [review]
LGTM but the profiler check is not needed...

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FnDefaultLanguage.java (hunk 1 above)
Comment (adamretter) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FnDefaultLanguage.java:49]: Ah right! So we could remove this from our `eval`. Also I just noticed, we should add an `@Override` to our eval

**Review by dizzzz — APPROVED** [review]

**Comment by joewiz** [comment]
@alanpaxton For good measure, could you please post the results of relevant XQTS tests for this function? Thanks!

**Pull request merged**